host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1289,6 +1367,74 @@ static void encoder_buffer_callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buf
 }
 
 /**
+ *  buffer header callback function for splitter
+ *
+ *  Callback will dump buffer data to the specific file
+ *
+ * @param port Pointer to port from which callback originated
+ * @param buffer mmal buffer header pointer
+ */
+static void splitter_buffer_callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buffer)
+{
+   MMAL_BUFFER_HEADER_T *new_buffer;
+   static int64_t base_time =  -1;
+
+   // All our times based on the receipt of the first callback
+   if (base_time == -1)
+      base_time = vcos_getmicrosecs64()/1000;
+
+   // We pass our file handle and other stuff in via the userdata field.
+
+   PORT_USERDATA *pData = (PORT_USERDATA *)port->userdata;
+
+   if (pData)
```

Comment:
Not a complaint about this review, but this lot really should get refactored as we now have it present about 4 times in the different callbacks.